src/memory/Array.hpp

```diff
@@ -33,12 +33,12 @@
  * @tparam M Allows to statically distinguish Arrays using different memory kinds.
  * @tparam T Type of elements
  */
-template<MemoryKind M, typename T>
-struct Array : public IAnyArray<M>
+template<typename T>
+struct Array : public IAnyArray
```

Comment:
M template parameter not removed from documentation.
There are also duplicated headers near the top of this file.